server/channels/app/post.go

```diff
@@ -774,109 +764,113 @@ func (a *App) UpdatePost(c request.CTX, receivedUpdatedPost *model.Post, safeUpd
 	return rpost, nil
 }
 
-func (a *App) publishWebsocketEventForPermalinkPost(c request.CTX, post *model.Post, message *model.WebSocketEvent) (published bool, err *model.AppError) {
-	var previewedPostID string
-	if val, ok := post.GetProp(model.PostPropsPreviewedPost).(string); ok {
-		previewedPostID = val
-	} else {
-		return false, nil
+func (a *App) publishWebsocketEventForPost(rctx request.CTX, post *model.Post, message *model.WebSocketEvent) *model.AppError {
+	postJSON, jsonErr := post.ToJSON()
+	if jsonErr != nil {
+		a.CountNotificationReason(model.NotificationStatusError, model.NotificationTypeAll, model.NotificationReasonMarshalError, model.NotificationNoPlatform)
+		a.NotificationsLog().Error("Error in marshalling post to JSON",
+			mlog.String("type", model.NotificationTypeWebsocket),
+			mlog.String("post_id", post.Id),
+			mlog.String("status", model.NotificationStatusError),
+			mlog.String("reason", model.NotificationReasonMarshalError),
+		)
+		return model.NewAppError("publishWebsocketEventForPost", "app.post.marshal.app_error", nil, "", http.StatusInternalServerError).Wrap(jsonErr)
+	}
+	message.Add("post", postJSON)
+
+	appErr := a.setupBroadcastHookForPermalink(rctx, post, message, postJSON)
+	if appErr != nil {
+		return appErr
+	}
+
+	a.Publish(message)
+	return nil
+}
+
+func (a *App) setupBroadcastHookForPermalink(rctx request.CTX, post *model.Post, message *model.WebSocketEvent, postJSON string) *model.AppError {
+	previewProp := post.GetPreviewedPostProp()
+	if previewProp == "" {
+		return nil
+	}
+
+	permalinkPreviewedPost := post.GetPreviewPost()
```

Comment:
What would happen if the prop is missing but the post has a preview post? It feels potentially brittle to have the decision decoupled.

Reply:
Sure, I could add another nil check here. Although I feel that things like this should be locked with unit tests so that other areas of the code (like here) can be confident that whenever a prop is there, the post has a preview post in metadata.

Reply:
Actually, my concern was more about leaking the data. Here we return early if the prop is missing but that doesn't guarantee there's no preview content, so if anything, I'd invert the checks or evaluate them both at the same time.

Reply:
Inverted the checks.